src/main/java/com/faforever/client/chat/ChatController.java

```diff
@@ -242,7 +242,7 @@ public void onDisplay(NavigateEvent navigateEvent) {
     super.onDisplay(navigateEvent);
     if (!tabPane.getTabs().isEmpty()) {
       Tab tab = tabPane.getSelectionModel().getSelectedItem();
-      nameToChatTabController.get(tab.getId()).onDisplay();
+      Optional.ofNullable(nameToChatTabController.get(tab.getId())).ifPresent(AbstractChatTabController::onDisplay);
```

Comment:
The real question is: How can we have a tab in the UI but no entry in `nameToChatTabController`?
That should not be possible, so here we're just concealing a bug.

Hint: synchronize access to `tabPane.getTabs()` and `nameToChatTabController` on `nameToChatTabController`.

Reply:
it happens if you lose connection

Reply:
The thing is it is possible that there is no tab selected on reconnect

Reply:
How is it possible that no tab is selected? To my understanding, if there are tabs then there is always exactly one selected.
The current implementation is not thread-safe. We have two lists and we assume that one list contains something the other list does as well, but this might change just after the first check.